File: ruby/bead.rb
#!/usr/bin/ruby

if not ARGV.length == 1
	puts "Adj meg egy könyvtárt!"
	exit
end

if not File.directory?(ARGV[0])
	puts "Nincs ilyen könyvtár!"
	exit
end

f1=File.open("result", 'w')

lel = Dir.entries(ARGV[0])
Dir.chdir(ARGV[0])
lel.each do |i|
	if ( i =~ /.rb/ )
		res = `ruby #{i}`
		if res=="1"
			f1.write(i+" :: OK\n")
		else
			f1.write(i+" :: WRONG\n")
		end
	end
end

f1.close

File: python/bead.py
#!/usr/bin/python

import sys
import os.path

if len(sys.argv)!=2:
	print "Adj meg egy file-t!"
else:
	if os.path.isfile(sys.argv[1]):
		f1=open(sys.argv[1], 'r')
		f2 = open('eredmeny.txt', 'w')
		data = f1.readline()
		max=-1
		ind=-1
		indyear=-1
		while data!='':
			data = f1.readline()
			datalist = data.split()
			if data!='':
				if int(datalist[6])>50:
					print datalist[6]
					f2.write(datalist[0] + ', ' + datalist[1] + ':: ' + datalist[7] + ' Ft\n')
				if int(max) < int(datalist[7]):
					max = datalist[7]
					indyear=datalist[0]
					ind = datalist[1]
		f2.write('max 4-es:: ' + indyear + ', ' + ind)
	else:
		print "Nincs ilyen file!"